From 48a5a087faf27bbc79745ca360009bb6ad74ff36 Mon Sep 17 00:00:00 2001
From: Nikita Travkin <nikita@trvn.ru>
Date: Mon, 16 Aug 2021 22:09:57 +0500
Subject: [PATCH 092/210] FIXME: ASoC: qcom: msm8916: enable jack detection
 kctls

msm8916 (apq8016) can detect the Headphones and External Mic presence
but userspace needs kctls to make use of that (e.g. with ucm
configurations)

Make those kctls available.

Signed-off-by: Nikita Travkin <nikita@trvn.ru>
---
 sound/soc/qcom/apq8016_sbc.c   | 3 +++
 sound/soc/qcom/msm8916-qdsp6.c | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/sound/soc/qcom/apq8016_sbc.c b/sound/soc/qcom/apq8016_sbc.c
index ba2a98268ee..0b1944e4673 100644
--- a/sound/soc/qcom/apq8016_sbc.c
+++ b/sound/soc/qcom/apq8016_sbc.c
@@ -105,6 +105,9 @@ static int apq8016_sbc_dai_init(struct snd_soc_pcm_runtime *rtd)
 		snd_jack_set_key(jack, SND_JACK_BTN_1, KEY_VOICECOMMAND);
 		snd_jack_set_key(jack, SND_JACK_BTN_2, KEY_VOLUMEUP);
 		snd_jack_set_key(jack, SND_JACK_BTN_3, KEY_VOLUMEDOWN);
+
+		snd_jack_add_new_kctl(jack, "Headphone", SND_JACK_HEADPHONE);
+		snd_jack_add_new_kctl(jack, "Headset Mic", SND_JACK_MICROPHONE);
 		pdata->jack_setup = true;
 	}
 
diff --git a/sound/soc/qcom/msm8916-qdsp6.c b/sound/soc/qcom/msm8916-qdsp6.c
index 92319dab646..b02ad99af7b 100644
--- a/sound/soc/qcom/msm8916-qdsp6.c
+++ b/sound/soc/qcom/msm8916-qdsp6.c
@@ -114,6 +114,10 @@ static int msm8916_qdsp6_dai_init(struct snd_soc_pcm_runtime *rtd)
 		snd_jack_set_key(jack, SND_JACK_BTN_1, KEY_VOICECOMMAND);
 		snd_jack_set_key(jack, SND_JACK_BTN_2, KEY_VOLUMEUP);
 		snd_jack_set_key(jack, SND_JACK_BTN_3, KEY_VOLUMEDOWN);
+
+		snd_jack_add_new_kctl(jack, "Headphone", SND_JACK_HEADPHONE);
+		snd_jack_add_new_kctl(jack, "Headset Mic", SND_JACK_MICROPHONE);
+
 		pdata->jack_setup = true;
 	}
 
-- 
2.34.1

